.github/scripts/distfiles.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+# Copyright (C) Viktor Szakats
+#
+# SPDX-License-Identifier: curl
+
+# Compare git repo files with tarball files and report a mismatch
+# after excluding exceptions.
+
+set -eu
+
+tarfiles="$(mktemp)"
+gitfiles="$(mktemp)"
+
+taronly="/
+^
+Makefile\.in
+^aclocal.m4
+^compile
+^configure
+^config\..+
+^depcomp
+^install-sh
+^ltmain\.sh
+^missing
+^docs/libcurl/libcurl-symbols\.md
+^docs/RELEASE-TOOLS\.md
+^docs/tarball-commit\.txt
+^lib/curl_config\.h\.in
+^m4/libtool\.m4
+^m4/lt.+\.m4
+^src/(tool_ca_embed|tool_hugehelp)\.c"
+
+gitonly="\.git.+
+^\..+
+^appveyor\..+
+^buildconf
+^renovate\.json
+^REUSE\.toml
+^(GIT-INFO|README|SECURITY)\.md
+^LICENSES/.+
+^docs/examples/adddocsref\.pl
+^docs/THANKS-filter
+^lib/optiontable\.pl
+^projects/Windows/.+
+^scripts/(ciconfig|cijobs|cmp-config|release-notes|singleuse)\.pl
+^scripts/(contributors|contrithanks|installcheck)\.sh
+^scripts/delta
+^src/tool_hugehelp\.c\.cvs
+^tests/CI\.md"
```

Comment:
Don't you think this list might be annoying to keep up-to-date going forward?

Reply:
Fair question, I think the list of exceptions is short, changes less often and is touched by few contributors (mainly you), when dealing with a few specific areas. Compared to the rest of files which get added more often by more people, related to any area. History shows it's easy to miss them [^1], and prone to cause fallouts.

To easy on the exceptions, we may aname/place them in such way that they are easy to exclude with more generic rules. E.g. by renaming `appveyor.*` to `.appveyor.*` or naming ci-specific scripts with `ci*.*`. There may be more.

[^1]: from the last 10 months:
b1153820f499cb511f04435f40ee463322816d8a
a4ad7dc5a3e21d573a02d8c0202e2bb26066f302
58946eed20dbf6f606c19ead5e34acb7aed817d8
5d3016adf7e855f03a644fb05308129806c98457
0a4419ae4c45fccaadc62dbdaf4725909c44dcfa
d0a851feadb28bb808ae9fe4e2c53a1636ec4c00
da8c1d15782c8161b455a7ee90197c16ae5edb90
fac90bd5c1cf81a3629979e2a7ec0757a916e213

Reply:
Would it make sense to move (some of) these inside `.github/scripts`?:
```
scripts/(ciconfig|cijobs|cmp-config|release-notes|singleuse)\.pl
```

Reply:
Except for `cmp-config`, they were all written for and I still occasionally use them outside of CI jobs so I think not.

And note: I don't deny that we have done mistakes. The question is only what the best way is to make us do less errors in the future without causing friction for everyone at the same time.

Reply:
Moved `cmp-config` there and dropped the exception.

Also deleted an earlier attempt that was limited to `./CMake/*.{cmake,in}` files.

Reply:
AFAICS the only extra friction is that non-distributable files now may need to be added to the exceptions.
In return, it guarantees that distributable ones are included in the tarball.

Reply:
Yeah, I can't figure out a better approach either.